tests/python/test_shared_array.py

```diff
@@ -46,9 +47,18 @@ def calc_shared_array(v: ti.types.ndarray(ndim=1),
     assert np.allclose(reference, a_arr)
 
 
-@test_utils.test(arch=[ti.cuda, ti.vulkan])
+@test_utils.test(arch=[ti.cpu, ti.cuda, ti.vulkan])
 def test_shared_array_nested_loop():
-    _test_shared_array_nested_loop()
+    arch = ti.lang.impl.get_runtime().prog.config().arch
```

Comment:
aha better split cuda/vulkan & cpu into different tests so that it's clear to expect throws in cpu.